src/Microsoft.SqlTools.ServiceLayer/BatchParser/BatchParserException.cs

```diff
@@ -0,0 +1,72 @@
+﻿//------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.SqlTools.ServiceLayer.BatchParser
+{
+    internal sealed class BatchParserException : Exception
+    {
+        const string ErrorCodeName = "ErrorCode";
+        const string BeginName = "Begin";
+        const string EndName = "End";
+        const string TextName = "Text";
+        const string TokenTypeName = "TokenType";
+
+        ErrorCode _errorCode;
+        PositionStruct _begin;
+        PositionStruct _end;
+        string _text;
+        LexerTokenType _tokenType;
+
+        public BatchParserException(ErrorCode errorCode, Token token, string message)
+            : base(message)
+        {
+            _errorCode = errorCode;
+            _begin = token.Begin;
+            _end = token.End;
+            _text = token.Text;
+            _tokenType = token.TokenType;
+        }
+
+        //leaving for incase we add serializable back again
```

Comment:
>//leaving for incase we add serializable back again [](start = 8, length = 51)

I would remove these commented out code blocks..it seems unlikely we'll add serialization back for this class and this just adds dead code that will likely be there indefinitely